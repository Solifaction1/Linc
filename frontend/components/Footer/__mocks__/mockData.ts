/* eslint-disable max-lines */
import { Contact } from "@models/contact.model"
import { Navigation, RelatedContactInformation } from "@models/navigation.model"

const mockNavigationParent1: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 1,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent1',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 1',
    type: '',
    uiRouterKey: 'parent1',
    updatedAt: ''
}

const mockNavigationParent2: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 4,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'parent2',
    related: null,
    sitemap_exclude: true,
    title: 'Parent 2',
    type: '',
    uiRouterKey: 'parent2',
    updatedAt: ''
}


export const mockContact: Contact = {
    id: 10,
    street: 'street',
    houseNumber: 'houseNumber',
    postalCode: 0,
    city: 'city',
    telephone: '+1234567890'
}

const mockContactInformationRelated: RelatedContactInformation = {
    createdAt: 'createdAt',
    facebook: 'facebook',
    id: 7,
    instagram: 'instagram',
    linkedin: 'linkedin',
    locale: 'en',
    navigationItemId: 0,
    publishedAt: 'publishedAt',
    sitemap_exclude: false,
    updatedAt: 'updatedAt',
    updatedBy: 'updatedBy',
    __contentType: '__contentType'
}

const mockContactInformation: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 8,
    items: null,
    menuAttached: true,
    order: 0,
    parent: null,
    path: 'contact-information',
    related: mockContactInformationRelated,
    sitemap_exclude: true,
    title: 'Contact Information',
    type: '',
    uiRouterKey: 'contact-information',
    updatedAt: ''
}

const mockNavigationDataCompliantLinks: Navigation = {
    audience: [],
    collapsed: true,
    createdAt: '',
    id: 2,
    items: null,
    menuAttached: true,
    order: 2,
    parent: null,
    path: 'data-compliant-links',
    related: null,
    sitemap_exclude: true,
    title: 'Data Compliant Links',
    type: '',
    uiRouterKey: 'data-compliant-links',
    updatedAt: ''
}

export const mockNavigationData = [
    mockNavigationParent1,
    mockNavigationParent2,
    mockContactInformation,
    mockNavigationDataCompliantLinks,
]