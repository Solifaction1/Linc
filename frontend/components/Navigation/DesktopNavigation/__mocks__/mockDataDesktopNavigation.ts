/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    items: 'items',
    parent: null,
    related: null
}]
