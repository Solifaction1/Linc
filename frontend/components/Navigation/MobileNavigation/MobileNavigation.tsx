/* eslint-disable max-lines */
import React, { useState, useEffect } from 'react'
import Image from 'next/image'
import Link from 'next/link'
import { Navigation } from '@models/navigation.model'
import LanguageSwitch from '@components/Language-switch/LanguageSwitch'
import { Localizations } from '@models/attributes.model'
import { getColor, getColorHeader } from '@constants/color'
import EmployeeData from '@models/employee.model'
import EmployeeSwitch from '@components/EmployeeSwitch/EmployeeSwitch'
import { getUniqueId } from '@utils/utils.service'
import styles from './mobileNavigation.module.scss'
import ContactItems from './ContactItems/ContactItems'

const MobileNavigation = ({
	internalLinksData,
	employeesData,
	className,
	contactButtonClassName,
	contactIconsClassName,
	localizations,
	color,
}: {
	internalLinksData: Navigation[]
	employeesData: { data: EmployeeData[] }
	className: string
	contactButtonClassName: string
	contactIconsClassName: string
	localizations: Localizations | undefined
	color: string | undefined
}) => {
	const [open, setOpen] = useState(false)

	useEffect(() => {
		const body = document.querySelector('body')
		if (body) body.style.overflow = open ? 'hidden' : 'auto'
	}, [open])

	const socialMediaItem = internalLinksData.find(
		(navigationitem) => navigationitem.path === 'social-media'
	)

	return (
		<div
			className={`${styles.navigationWrapper} ${
				open && styles.open
			} ${className} ${styles[getColor(color ?? '')]}`}
			role="button"
			tabIndex={0}
		>
			<header className={styles.navigationHeader}>
				<Link href="/">
					{color === 'beige' || color === 'blue-gray' || color === 'white' ? (
						<Image
							src="/images/logo-gray.svg"
							alt="logo"
							width={112}
							height={44}
							className={styles.logo}
							priority
							quality={60}
						/>
					) : (
						<Image
							src="/images/logo.svg"
							alt="logo"
							width={112}
							height={44}
							className={styles.logo}
							priority
							quality={60}
						/>
					)}
				</Link>
				<button
					type="button"
					onClick={() => setOpen((prevstate) => !prevstate)}
					className={`${styles.hamburgerIcon} ${styles[getColor(color ?? '')]}`}
				>
					<div className={open ? styles.open : ''} />
					<div className={open ? styles.open : ''} />
					<div className={open ? styles.open : ''} />
				</button>
			</header>
			<ul
				className={`${styles.navigationContainer} ${className} ${
					open && styles.open
				}`}
			>
				{internalLinksData.map((item) => {
					if (item.path === 'contact') {
						return (
							<ContactItems
								contactItem={item}
								socialMediaItem={socialMediaItem}
								contactButtonClassName={contactButtonClassName}
								contactIconsClassName={contactIconsClassName}
								color={`${getColorHeader(color ?? '')}`}
								key={item.id}
							/>
						)
					}

					if (item.path === 'social-media') return ''

					if (item.path === 'team') {
						return (
							<EmployeeSwitch
								parent={item}
								employeeData={employeesData}
								key={getUniqueId()}
								className={styles.navigationMainLinkWrapper}
							/>
						)
					}

					return (
						<div key={item.id} className={styles.navigationMainLinkWrapper}>
							<Link
								href={`/${item?.path}`}
								replace
								className={styles.navigationMainLink}
							>
								<li>{item.title}</li>
							</Link>
						</div>
					)
				})}

				<div className={styles.languageSwitch}>
					<LanguageSwitch localizations={localizations} />
				</div>
			</ul>
		</div>
	)
}

export default MobileNavigation
