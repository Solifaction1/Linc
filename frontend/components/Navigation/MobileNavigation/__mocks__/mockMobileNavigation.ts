/* eslint-disable max-lines */
import { Navigation } from "@models/navigation.model";

export const navigationMockData: Navigation[] = [{
    audience: [],
    collapsed: false,
    createdAt: 'createdAt',
    id: 1,
    menuAttached: false,
    order: 0,
    path: 'path',
    sitemap_exclude: false,
    title: 'title',
    type: 'type',
    uiRouterKey: 'uiRouterKey',
    updatedAt: 'updatedAt',
    children: [],
    items: 'items',
    parent: null,
    related: {
        createdAt: '',
        facebook: 'https://www.facebook.com',
        linkedin: 'https://www.linkedin.com',
        instagram: '',
        locale: '',
        id: 1,
        navigationItemId: 2,
        publishedAt: '',
        sitemap_exclude: false,
        updatedAt: '',
        updatedBy: '',
        __contentType: ''
    }
}]
