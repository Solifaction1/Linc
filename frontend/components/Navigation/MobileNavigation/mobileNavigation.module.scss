@import 'styles/vars';
@import 'styles/utilities';

.navigationWrapper {
	position: fixed;
	overflow: hidden;

	width: 100%;
	height: 100vh;
	max-height: 100dvh; 
	
	pointer-events: none;

	&.open {
		pointer-events: all;
	}

	.navigationHeader {
		z-index: 1005;
		display: flex;
		align-items: center;
		justify-content: space-between;

		width: 100%;
		height: $header-height;
		padding: calc(#{$spacing-base} * 3) $page-bleed calc(#{$spacing-base} * 2);

		@media (max-width: $break-s) {
			padding: calc(#{$spacing-base} * 3) $page-bleed-mobile
				calc(#{$spacing-base} * 2);
		}

		a {
			height: fit-content;
			pointer-events: all;
		}

		.hamburgerIcon {
			position: relative;

			width: 5rem;
			height: 5rem;
			border: none;

			background-color: transparent;

			cursor: pointer;
			pointer-events: all;

			&.beige,
			&.blueGray {
				div {
					background-color: $color-gray;
				}
			}

			&.white {
				div {
					background-color: $color-black;
				}
			}

			div {
				content: '';
				position: absolute;
				right: 50%;
				bottom: 60%;

				width: 1.5rem;
				height: $border-width-md;
				border-radius: 0.3rem;

				background-color: $color-white;

				transform: translate(50%, 50%);
				transform-origin: center;

				transition: transform 0.2s ease;

				&:nth-child(1) {
					transform: translate(50%, calc(50% - 0.5rem));

					&.open {
						transform: translate(50%, 50%) rotate(-45deg);
					}
				}

				&:nth-child(2) {
					&.open {
						transform: translate(50%, 50%) rotate(-45deg);
					}
				}

				&:nth-child(3) {
					right: 55%;
					width: 0.94rem;
					transform: translate(50%, calc(50% + 0.5rem));

					&.open {
						right: 50%;
						width: 1.5rem;
						transform: translate(50%, 50%) rotate(45deg);
					}
				}
			}
		}
	}

	.navigationContainer {
		top: $header-height;
		z-index: 1004;
		display: flex;
		flex-direction: column;
		overflow: auto;

		width: 100%;
		height: calc(100vh);
		padding: 2rem $page-bleed;

		transform: translateX(100%);

		transition: transform $animation-in-lg ease;

		&.open {
			transform: translateY(0);
		}

		.navigationMainLinkWrapper {
			position: relative;

			width: 100%;
			padding: calc(#{$spacing-base} * 2) 0;

			background: none;

			&.underline {
				padding-bottom: calc(#{$spacing-base} * 4);
				border-bottom: $border-width-sm solid rgba(40, 39, 34, 16%);
			}

			.navigationMainLink {
				display: flex;
				justify-content: space-between;

				width: 100%;
				padding: calc(#{$spacing-base} * 1) 0;
				border: none;

				background: none;

				cursor: pointer;

				.arrow {
					transition: transform $animation-in ease;

					&.active {
						transform: rotate(-180deg);
					}
				}

				li {
					font-size: $font-size-title-md-m;
					line-height: 3.5rem;
					font-weight: $font-weight-regular;
				}
			}
		}

		.languageSwitch {
			button {
				padding-left: 0;
			}
		}
	}

	&.gray {
		.navigationHeader,
		.navigationContainer {
			background-color: $color-gray;
		}
	}

	&.beige {
		.navigationHeader,
		.navigationContainer {
			background-color: $color-beige;
		}
	}

	&.blueGray {
		.navigationHeader,
		.navigationContainer {
			background-color: $color-blue-gray;
		}
	}

	&.darkGreen {
		.navigationHeader,
		.navigationContainer {
			background-color: $color-green-dark;
		}
	}

	&.lightGreen {
		.navigationHeader,
		.navigationContainer {
			background-color: $color-green;
		}
	}

	&.white {
		.navigationHeader,
		.navigationContainer {
			background-color: $color-white;
		}
	}
}
