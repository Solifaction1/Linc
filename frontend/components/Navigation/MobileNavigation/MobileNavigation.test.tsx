import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'
import MobileNavigation from './MobileNavigation'
import { navigationMockData } from './__mocks__/mockMobileNavigation'

jest.mock('next/router', () => ({
	useRouter() {
		return {
			locale: 'locale',
		}
	},
}))

describe('MobileNavgation', () => {
	it('renders a MobileNavgation', () => {
		const { container, getByText } = render(
			<MobileNavigation
				internalLinksData={navigationMockData}
				className="className"
				localizations={{ data: undefined }}
				color="dark-green"
				contactButtonClassName="contactButtonClassName"
				contactIconsClassName="contactIconsClassName"
				employeesData={{ data: [] }}
			/>
		)

		expect(container).toMatchSnapshot()

		expect(
			container.getElementsByClassName(
				'navigationWrapper false className darkGreen'
			)[0]
		).toBeVisible()

		expect(container.querySelector('header')).toBeVisible()
		expect(
			container.getElementsByClassName('navigationHeader')[0]
		).toBeVisible()
		expect(container.querySelector('a[href="/"]')).toBeVisible()

		expect(container.querySelector('img')).toBeVisible()
		expect(container.getElementsByClassName('logo')[0]).toBeVisible()

		expect(container.getElementsByClassName('hamburgerIcon')[0]).toBeVisible()

		expect(
			container.getElementsByClassName('navigationContainer className false')[0]
		).toBeVisible()
		expect(
			container.getElementsByClassName('navigationMainLinkWrapper')[0]
		).toBeVisible()

		expect(
			container.getElementsByClassName('navigationMainLink')[0]
		).toBeVisible()
		expect(container.querySelector('a[href="/path"]')).toBeVisible()
		expect(getByText('title', { selector: 'li' })).toBeVisible()

		expect(container.getElementsByClassName('languageSwitch')[0]).toBeVisible()

		expect(getByText('LOCALE', { selector: 'button' })).toBeVisible()
	})
})
