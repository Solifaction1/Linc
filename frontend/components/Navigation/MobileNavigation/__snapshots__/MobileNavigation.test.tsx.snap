// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNavgation renders a MobileNavgation 1`] = `
<div>
  <div
    class="navigationWrapper false className darkGreen"
    role="button"
    tabindex="0"
  >
    <header
      class="navigationHeader"
    >
      <a
        href="/"
      >
        <img
          alt="logo"
          class="logo"
          data-nimg="1"
          decoding="async"
          height="44"
          src="/images/logo.svg"
          srcset="/images/logo.svg 1x, /images/logo.svg 2x"
          style="color: transparent;"
          width="112"
        />
      </a>
      <button
        class="hamburgerIcon darkGreen"
        type="button"
      >
        <div
          class=""
        />
        <div
          class=""
        />
        <div
          class=""
        />
      </button>
    </header>
    <ul
      class="navigationContainer className false"
    >
      <div
        class="navigationMainLinkWrapper"
      >
        <a
          class="navigationMainLink"
          href="/path"
        >
          <li>
            title
          </li>
        </a>
      </div>
      <div
        class="languageSwitch"
      >
        <div
          class="languageSwitch"
        >
          <button
            class="languageLink activeLang"
            type="button"
          >
            LOCALE
            <svg
              class="false arrow"
              fill="currentColor"
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M17,9.17a1,1,0,0,0-1.41,0L12,12.71,8.46,9.17a1,1,0,0,0-1.41,0,1,1,0,0,0,0,1.42l4.24,4.24a1,1,0,0,0,1.42,0L17,10.59A1,1,0,0,0,17,9.17Z"
              />
            </svg>
          </button>
          <div
            class="accordionWrapper"
            style="height: 0px;"
          >
            <div
              class="languageLinkWrapper"
            />
          </div>
        </div>
      </div>
    </ul>
  </div>
</div>
`;
